editoast/editoast_schemas/src/rolling_stock/etcs_brake_params.rs

```diff
@@ -0,0 +1,115 @@
+use derivative::Derivative;
+use serde::Deserialize;
+use serde::Deserializer;
+use serde::Serialize;
+use utoipa::ToSchema;
+
+editoast_common::schemas! {
+    EtcsBrakeParams,
+    SpeedIntervalValueCurve,
+}
+
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize, ToSchema, Derivative)]
+#[derivative(Hash)]
+#[serde(deny_unknown_fields)]
+/// Braking parameters for ERTMS ETCS Level 2
+pub struct EtcsBrakeParams {
+    /// A_brake_emergency: the emergency deceleration curve (values in m/s²)
+    pub gamma_emergency: SpeedIntervalValueCurve,
+    /// A_brake_service: the full service deceleration curve (values in m/s²)
+    pub gamma_service: SpeedIntervalValueCurve,
+    /// A_brake_normal_service: the normal service deceleration curve used to compute guidance curve (values in m/s²)
+    pub gamma_normal_service: SpeedIntervalValueCurve,
+    /// Kdry_rst: the rolling stock deceleration correction factors for dry rails
+    /// Boundaries should be the same as gammaEmergency
+    /// Values (no unit) should be contained in [0, 1]
+    pub k_dry: SpeedIntervalValueCurve,
+    /// Kwet_rst: the rolling stock deceleration correction factors for wet rails
+    /// Boundaries should be the same as gammaEmergency
+    /// Values (no unit) should be contained in [0, 1]
+    pub k_wet: SpeedIntervalValueCurve,
+    /// Kn+: the correction acceleration factor on normal service deceleration in positive gradients
+    /// Values (in m/s²) should be contained in [0, 10]
+    pub k_n_pos: SpeedIntervalValueCurve,
+    /// Kn-: the correction acceleration factor on normal service deceleration in negative gradients
+    /// Values (in m/s²) should be contained in [0, 10]
+    pub k_n_neg: SpeedIntervalValueCurve,
+    /// T_traction_cut_off: time delay in s from the traction cut-off command to the moment the acceleration due to traction is zero
+    #[derivative(Hash(hash_with = "editoast_common::hash_float::<5,_>"))]
+    pub t_traction_cut_off: f64,
+    /// T_bs1: time service break in s used for SBI1 computation
+    #[derivative(Hash(hash_with = "editoast_common::hash_float::<5,_>"))]
+    pub t_bs1: f64,
+    /// T_bs2: time service break in s used for SBI2 computation
+    #[derivative(Hash(hash_with = "editoast_common::hash_float::<5,_>"))]
+    pub t_bs2: f64,
+    /// T_be: safe brake build up time in s
+    #[derivative(Hash(hash_with = "editoast_common::hash_float::<5,_>"))]
+    pub t_be: f64,
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, ToSchema, Derivative)]
+#[derivative(Hash)]
+#[serde(deny_unknown_fields)]
+pub struct SpeedIntervalValueCurve {
+    #[derivative(Hash(hash_with = "editoast_common::hash_float_slice::<3,_>"))]
+    #[schema(example = json!([8.333333, 19.444444]))]
+    /// Speed in m/s (sorted)
```

Comment:
nitpick: sorted in which order?

```suggestion
    /// Speed in m/s (sorted from smallest to biggest)
```

Reply:
Done in https://github.com/OpenRailAssociation/osrd/pull/10147/commits/983dc60fa77a8739dea21dc67bb93a2a03f10265